packages/selection/src/useSelection.ts

```diff
@@ -5,141 +5,154 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import { FocusEventHandler, KeyboardEventHandler, useEffect, useReducer } from 'react';
+import {
+  FocusEventHandler,
+  KeyboardEventHandler,
+  MutableRefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer
+} from 'react';
 import { composeEventHandlers, getControlledValue, KEYS } from '@zendeskgarden/container-utilities';
 import { IUseSelectionProps, IUseSelectionReturnValue } from './types';
 import { stateReducer } from './utils';
 
-export const useSelection = <Item>({
+export const useSelection = <Value>({
+  values,
   direction = 'horizontal',
-  defaultFocusedIndex = 0,
-  defaultSelectedIndex,
+  defaultFocusedValue = values[0],
+  defaultSelectedValue,
   rtl,
-  selectedItem,
-  focusedItem,
+  selectedValue,
+  focusedValue,
   onSelect,
   onFocus
-}: IUseSelectionProps<Item> = {}): IUseSelectionReturnValue<Item> => {
-  const isSelectedItemControlled = selectedItem !== undefined;
-  const isFocusedItemControlled = focusedItem !== undefined;
-  const refs: React.MutableRefObject<any | null>[] = [];
-  const items: Item[] = [];
+}: IUseSelectionProps<Value>): IUseSelectionReturnValue<Value> => {
+  const isSelectedValueControlled = selectedValue !== undefined;
+  const isFocusedValueControlled = focusedValue !== undefined;
+
+  // Create a ref dictionary from `values`.
+  // Refs are created/assigned as part of `getElementProps`.
+  const refs = useMemo(
+    () =>
+      values.reduce((all: Record<string, MutableRefObject<any | null>>, value) => {
+        all[String(value)] = { current: null };
```

Comment:
I think this `String` coercion is going to be problematic for object values. Can't the `value` just be used as the key?